### ISSUE TITLE
[no-unused-vars] Incorrect reporting when using spread

### ISSUE THREAD

**Issue by arcanis** [typescript-eslint/typescript-eslint#763]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [2]
  }
}
```

```TS
const {foo, ...rest} = bar;
doSomething(rest);
```

**Expected Result**

Shouldn't generate an error.

**Actual Result**

Generates an error:

```
  1:8  error  'foo' is assigned a value but never used  @typescript-eslint/no-unused-vars
```

**Additional Info**

ESLint isn't wrong in that `foo` is, indeed, unused. However, because its sheer presence has a side effect on `rest`, it shouldn't be marked unused if `rest` is used.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0-alpha` |
| `ESLint`                           | `6.1.0` |

**Comment by bradzacher** [comment]
Try this option:
https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings
`"@typescript-eslint/no-unused-vars": ["error", { "ignoreRestSiblings": true }]`

[demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tdW51c2VkLXZhcnM6IFtcImVycm9yXCIsIHsgXCJpZ25vcmVSZXN0U2libGluZ3NcIjogdHJ1ZSB9XSovXG5jb25zdCB7Zm9vLCAuLi5yZXN0fSA9IGJhcjtcbmRvU29tZXRoaW5nKHJlc3QpO1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo5LCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6e30sImVudiI6eyJjb21tb25qcyI6ZmFsc2UsIm5vZGUiOmZhbHNlfX19)

**Comment by Robloche** [comment]
This option does not work for me...

My code is:
`
for (const key of keys) {
    const {[key]: val} = map;
    ...
}
`
